djangocms_references/helpers.py

```diff
@@ -92,6 +158,12 @@ def contenttype_values_queryset(queryset):
 
 
 def convert_plugin_querysets_to_sources(querysets):
+    """Convert provided plugin querysets to a form containing
+    only source object information and concatenate them into one
+    using UNION.
```

Comment:
I'm still confused about what this returns exactly. "source objects" could potentially be anything and I don't think you could really combine any type of queryset using union? So are you returning a CMSPlugin queryset?